### PR TITLE
Unify unit tests

### PR DESCRIPTION
**Pull request by white238**
Unifying unit test executables to try to cut down on the 1.7 gig of executables in our tests directory

### CODE REVIEW

File: src/axom/core/tests/core_stack_array.hpp [hunk 1]
```diff
@@ -112,7 +112,7 @@ std::ostream& operator<<(std::ostream& stream, const Tensor& t)
   return stream;
 }
 
-TEST(StackArray, storage)
+TEST(core_stack_array, storage)
```
Comment (kennyweiss) [inline src/axom/core/tests/core_stack_array.hpp:115]: 👍

File: src/axom/sidre/tests/spio/spio_main.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "gtest/gtest.h"
+
+#include "mpi.h"
+
```
Comment (kennyweiss) [inline src/axom/sidre/tests/spio/spio_main.cpp:9]: This needs `#include axom/config.hpp` to work

**Review by kennyweiss — APPROVED** [review]
Thanks @white238

**Comment by white238** [comment]
Going to break these PRs up into several so its not 200+ files in one...

**Review by agcapps — APPROVED** [review]
Bravely done, @white238 .  Thank you.

**Pull request merged**